docs/rfc/derived-aspects/derived-aspects.md

```diff
@@ -0,0 +1,167 @@
+
+#### Get
+1. Invoke `ValueComputer` and compute the current value of the derived aspect.
+2. Check if the derived metadata has changed from previous persisted value n KV store.
+3. Persist if the metadata has changed.
+4. Retrieve latest value of derived metadata and serve the `GET` request.
+
+The persistence of the derived aspect follows GMA architecture. This helps with few of the motivational points mentioned.
+1. Versioning capabilities of the derived metadata
+2. Notification in the form of MAE when derived metadata has changed. With this, clients of this API need not ping GMS every other day to know if the compliance has changed.
+
+The MAEs produced will further power both search and graph GMA indices. This is similar to any other GMA aspect.
+
+If there is a process setup for dumping mysql store to analytical systems like HDFS, derived aspects also get naturally dumped along with other regular aspects.
+
+### Refresh of derived aspect metadata
+`GET` operation on derived aspect always gives the current state of derived metadata. 
+
+Often, `GET` operation might not be called for all entity values. Hence, peeking into the DB doesn't guarantee to give the true state of derived aspect.
+   
+Depending on how fresh we need GMA indexes and how often we want to notify on the changes of derived aspect metadata, there is a need for a CRON job which calls GET operation for all URNs of the entity.  
+
+ ![Backfill Derived Aspects](Backfill-Process.png)
+
+In the above, a CRON job(or a regular backfill job) on a nightly basis calls `GET` API of derived aspect.
+This ensures a few things
+   - KV store is fresh after every 24 hours
+   - Notification of change in derived metadata happens at-least once in 24 hours. 
+       - Note that some other clients can call `GET` API with in 24 hours and there is a possibility of emitting MAE more than once in a day.
+
+There might be cases derived aspects can be computed in realtime based on notifications (e.g., MAEs of different aspects).
+In these cases, a CRON job is technically not necessary.
+However, having a CRON job will ensure to have a process of maintaining freshness of the data even if a notification got missed while processing.
+
+## How we teach this
+
+- Derived metadata gets derived from multiple aspects of possibly different(or same) entities with(or without) any other auxiliary information.
+- Derived metadata can be modelled as aspects and are called as `Derived Aspects`. These are very similar to any other GMA aspect.
+- `GET` API of `Derived Aspect` always gives the up-to date derived metadata.
+- `get` calls on DAOs are supported and `add` calls can be blocked. Clients will only declare computation method of the derived aspect as a `ValueComputer` class.
+- A CRON job maintains the freshness of the `Derived Aspect` persisted in KV store and GMA indices.
+
+## Drawbacks
```

Comment:
- Lazy architecture means derived data is out of date until retrieved; causing a "delay" in MAEs. May not be suitable for all use cases.
- We're adding a lot of new stuff here for this very specific use case that will be hard to extent/adapt/change in the future.